## To complete this challenge, determine the five most common journals and the total articles for each. Next, calculate the mean, median, and standard deviation of the open-access cost per article for each journal.

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import pylab 
import scipy.stats as stats
import os

In [3]:
os.getcwd()

'C:\\Users\\gothv\\Practice\\Thinkful- Assignments'

In [4]:
os.chdir('C:\\Users\\gothv\\OneDrive\\Desktop\\Data\\Datasets')

In [5]:
os.getcwd()

'C:\\Users\\gothv\\OneDrive\\Desktop\\Data\\Datasets'

In [6]:
df= pd.read_csv('WELLCOME_APCspend2013_forThinkful.csv', encoding = "ISO-8859-1")

Note to self: look into what encoding errors exist and why!

In [7]:
df.head()

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [8]:
df.columns

Index(['PMID/PMCID', 'Publisher', 'Journal title', 'Article title',
       'COST (£) charged to Wellcome (inc VAT when charged)'],
      dtype='object')

In [9]:
df.dtypes

PMID/PMCID                                             object
Publisher                                              object
Journal title                                          object
Article title                                          object
COST (£) charged to Wellcome (inc VAT when charged)    object
dtype: object

In [11]:
df.describe()

PMID/PMCID Publisher Journal title  \
count                1928      2127          2126   
unique               1880       299           984   
top     Not yet available  Elsevier      PLoS One   
freq                    7       387            92   

                                            Article title  \
count                                                2127   
unique                                               2126   
top     Exclusive breastfeeding, diarrhoel morbidity a...   
freq                                                    2   

       COST (£) charged to Wellcome (inc VAT when charged)  
count                                                2127   
unique                                               1402   
top                                              £2040.00   
freq                                                   94

In [13]:
[name.lower().replace(' ', "_") for name in list (df.columns)]

['pmid/pmcid',
 'publisher',
 'journal_title',
 'article_title',
 'cost_(£)_charged_to_wellcome_(inc_vat_when_charged)']

In [24]:
new_cols = [name.lower().replace(' ', "_") for name in list (df.columns)]

In [25]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       article_title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  cost_(£)_charged_to_wellcome_(inc_vat_when_charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [15]:
repl_cols_dict = dict(zip(list(df.columns), new_cols))

In [16]:
df.rename(columns = repl_cols_dict, inplace = True)

In [17]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       article_title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  cost_(£)_charged_to_wellcome_(inc_vat_when_charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

In [18]:
df.isnull().sum()

pmid/pmcid                                             199
publisher                                                0
journal_title                                            1
article_title                                            0
cost_(£)_charged_to_wellcome_(inc_vat_when_charged)      0
dtype: int64

In [19]:
df.dropna(subset= ['journal_title'], inplace= True)

In [20]:
df.isnull().sum()

pmid/pmcid                                             198
publisher                                                0
journal_title                                            0
article_title                                            0
cost_(£)_charged_to_wellcome_(inc_vat_when_charged)      0
dtype: int64

In [21]:
pd.to_numeric(df['cost_(£)_charged_to_wellcome_(inc_vat_when_charged)'], errors='coerce');

In [22]:
df.groupby(['journal_title']);

In [23]:
df['journal_title'].value_counts()[0:5]

PLoS One                                           92
PLoS ONE                                           62
Journal of Biological Chemistry                    48
Nucleic Acids Research                             21
Proceedings of the National Academy of Sciences    19
Name: journal_title, dtype: int64

In [26]:
df['journal_title'] = df['journal_title'].str.lower()

In [27]:
df['journal_title'].value_counts()[0:5]

plos one                           190
journal of biological chemistry     53
neuroimage                          29
plos genetics                       24
plos pathogens                      24
Name: journal_title, dtype: int64

It looks like the top 5 journals are 

## Results for top five journals:
1. **PLoS One:** 190 total articles
2. **Journal of Biological Chemistry:** 53 total articles
3. **Neuroimage:** 29 total articles
4. **Plos Genetics:** 24 total articles
5.  **Plos Pathogens:** 24 total articles

# Part 2: Calculating mean, median, and standard deviation of open-access cost of access for each journal.


In [30]:
df.head(100)

pmid/pmcid  \
0                                                 NaN   
1                                          PMC3679557   
2                               23043264  PMC3506128    
3                                 23438330 PMC3646402   
4                                23438216 PMC3601604    
5                                          PMC3579457   
6                                          PMC3709265   
7                                23057412 PMC3495574    
8                                   PMCID: PMC3780468   
9                                   PMCID: PMC3621575   
10                                  PMCID: PMC3739413   
11                                  PMCID: PMC3530961   
12                                  PMCID: PMC3624797   
13                                         PMC3413243   
14                                         PMC3694353   
15                                         PMC3572711   
16                                           22610094   
17                                  PMCID: PMC3586974   
18                       23455506  PMCID: PMC3607399    
19                         PMID: 24015914 PMC3833349    
20                                      : PMC3805332    
21                                                NaN   
22                           PMCID:\n    PMC3656742\n   
23                                     PMCID: 3584654   
24                                22971149 PMC3466778   
25                                           23072351   
26                                         PMC3477822   
27                                         PMC3530960   
28                               23458425 PMC3630740    
29                                     PMID: 20146481   
..                                                ...   
70                                  PMCID: PMC3724652   
71                           PMCID:\n    PMC3597794\n   
72                                         PMC3522294   
73                                         PMC3774408   
74                                         PMC3610987   
75                                            3610976   
76                                     PMC3696691\n\n   
77                              23344959  PMC3585060    
78                                  PMCID: PMC3481303   
79                             PMCID:\n    PMC3789947   
80                                 PMCID: PMC3663556    
81                                            3947303   
82                         PMID: 23856774 PMC3811480    
83                                  PMCID: PMC3553740   
84                   PMID:23817378  PMCID: PMC3754341   
85                                           23877697   
86                                 PMCID: PMC3675966    
87                                         PMC3697472   
88                                         PMC3571300   
89                          PMID:22988015 PMC3497410    
90                                    PMC in progress   
91                                  PMCID: PMC3497412   
92                                         PMC3754575   
93                                                NaN   
94                          PMID: 23152534 PMC3554137   
95                           PMCID:\n    PMC3648151\n   
96                         PMCID:\n    PMC3518918\n\n   
97                                  PMCID: PMC3624257   
98                             PMCID:\n    PMC3697669   
99  PMCID:\n    PMC3554100\n    [Available on 2013...   

                                            publisher  \
0                                                 CUP   
1                                                 ACS   
2                                                 ACS   
3                                                 ACS   
4                                                 ACS   
5                                                 ACS   
6                                                 ACS   
7                                                 AC

In [32]:
df.rename(columns = {'cost_(£)_charged_to_wellcome_(inc_vat_when_charged)': 'cost'}, inplace= True)

In [33]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  psychological medicine   
1             PMC3679557       ACS       biomacromolecules   
2  23043264  PMC3506128        ACS              j med chem   
3    23438330 PMC3646402       ACS              j med chem   
4   23438216 PMC3601604        ACS              j org chem   

                                       article_title      cost  
0  Reduced parahippocampal cortical thickness in ...     £0.00  
1  Structural characterization of a Model Gram-ne...  £2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...   £642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...   £669.64  
4  Regioselective opening of myo-inositol orthoes...   £685.88

In [37]:
df['cost']= df['cost'].str.replace('£', ' ')

In [38]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  psychological medicine   
1             PMC3679557       ACS       biomacromolecules   
2  23043264  PMC3506128        ACS              j med chem   
3    23438330 PMC3646402       ACS              j med chem   
4   23438216 PMC3601604        ACS              j org chem   

                                       article_title      cost  
0  Reduced parahippocampal cortical thickness in ...      0.00  
1  Structural characterization of a Model Gram-ne...   2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...    642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...    669.64  
4  Regioselective opening of myo-inositol orthoes...    685.88

In [39]:
df.dtypes

pmid/pmcid       object
publisher        object
journal_title    object
article_title    object
cost             object
dtype: object

In [45]:
df['cost']= df['cost'].str.replace('$', ' ')

In [46]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  psychological medicine   
1             PMC3679557       ACS       biomacromolecules   
2  23043264  PMC3506128        ACS              j med chem   
3    23438330 PMC3646402       ACS              j med chem   
4   23438216 PMC3601604        ACS              j org chem   

                                       article_title      cost  
0  Reduced parahippocampal cortical thickness in ...      0.00  
1  Structural characterization of a Model Gram-ne...   2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...    642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...    669.64  
4  Regioselective opening of myo-inositol orthoes...    685.88

In [47]:
df['cost']= df['cost'].astype(float)

In [48]:
df.head()

pmid/pmcid publisher           journal_title  \
0                    NaN       CUP  psychological medicine   
1             PMC3679557       ACS       biomacromolecules   
2  23043264  PMC3506128        ACS              j med chem   
3    23438330 PMC3646402       ACS              j med chem   
4   23438216 PMC3601604        ACS              j org chem   

                                       article_title     cost  
0  Reduced parahippocampal cortical thickness in ...     0.00  
1  Structural characterization of a Model Gram-ne...  2381.04  
2  Fumaroylamino-4,5-epoxymorphinans and related ...   642.56  
3  Orvinols with mixed kappa/mu opioid receptor a...   669.64  
4  Regioselective opening of myo-inositol orthoes...   685.88

In [52]:
df.dtypes

pmid/pmcid        object
publisher         object
journal_title     object
article_title     object
cost             float64
dtype: object

In [73]:
plos_df =df.loc[df['journal_title']== 'plos one']

In [74]:
plos_df.head()

pmid/pmcid publisher journal_title  \
1282    3517619      PLoS      plos one   
1283    3498109      PLoS      plos one   
1284    3515553      PLoS      plos one   
1285    3522679      PLoS      plos one   
1286    3485223      PLoS      plos one   

                                          article_title     cost  
1282  HCN1 and HCN2 in Rat DRG Neurons: Levels in No...  1001.03  
1283  Fetal alcohol exposure and IQ at age 8: Eviden...  1004.15  
1284  Vitamin B-12 status during pregnancy and child...  1011.45  
1285  Validation of Dual Energy X-ray Absorptiometry...  1011.45  
1286  Associations of different phenotypes of wheezi...  1015.73

In [87]:
plos_df.shape

(190, 5)

In [100]:
round(plos_df.describe(), 2)

cost
count     190.00
mean    38751.29
std    189009.51
min       122.31
25%       852.46
50%       898.48
75%      1044.74
max    999999.00

In [82]:
bio_chem = df.loc[df['journal_title'] == 'journal of biological chemistry']

In [84]:
bio_chem.shape

(53, 5)

In [99]:
round(bio_chem.describe(), 2)

cost
count      53.00
mean    20264.63
std    137165.49
min       265.67
25%      1152.72
50%      1314.53
75%      1610.31
max    999999.00

In [88]:
neuro= df.loc[df['journal_title'] == 'neuroimage']

In [89]:
neuro.shape

(29, 5)

In [98]:
round(neuro.describe(), 2)

cost
count    29.00
mean   2215.17
std     266.65
min    1747.16
25%    2100.54
50%    2326.43
75%    2396.57
max    2503.34

In [93]:
genetics= df.loc[df['journal_title'] == 'plos genetics']

In [94]:
genetics.shape

(24, 5)

In [97]:
round(genetics.describe(), 2)

cost
count      24.00
mean    84839.44
std    281865.71
min      1394.05
25%      1484.86
50%      1718.39
75%      1778.22
max    999999.00

In [101]:
patho= df.loc[df['journal_title']== 'plos pathogens']

In [102]:
patho.shape

(24, 5)

In [103]:
round(patho.describe(), 2)

cost
count      24.00
mean    84775.04
std    281885.54
min      1254.02
25%      1440.00
50%      1600.52
75%      1760.31
max    999999.00